metrics/firebase/functions/test/main_test.dart

```diff
@@ -0,0 +1,250 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:firebase_functions_interop/firebase_functions_interop.dart';
+import 'package:metrics_core/metrics_core.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../lib/main.dart';
+import '../lib/mappers/build_day_status_field_name_mapper.dart';
+
+void main() {
+  group(("onBuildAddedHandler"), () {
+    const buildDaysCollectionName = 'build_days';
+    const tasksCollectionName = 'tasks';
+    const projectId = 'projectId';
+    const durationInMilliseconds = 123;
+    final utcDate = _getDateInUTC(DateTime.now());
+    final buildStatus = BuildStatus.successful;
+
+    final build = {
+      'duration': durationInMilliseconds,
+      'projectId': projectId,
+      'buildStatus': buildStatus.toString(),
+      'startedAt': Timestamp.fromDateTime(utcDate),
+      'coverage': 100,
+      'url': 'url',
+      'workflowName': 'workflowName',
+    };
+
+    final mapper = BuildDayStatusFieldNameMapper();
+    final buildDayStatusFieldName = mapper.map(buildStatus);
+
+    final expectedBuildDayData = {
+      'projectId': projectId,
+      buildDayStatusFieldName: 1,
+      'totalDuration': durationInMilliseconds,
+      'day': Timestamp.fromDateTime(utcDate),
+    };
+
+    final exception = Exception('test');
+    final expectedCreatedTask = {
+      'code': 'build_days_created',
+      'data': build,
+      'context': exception.toString(),
+      'createdAt': Timestamp.fromDateTime(utcDate),
+    };
+
+    final _firestoreMock = FirestoreMock();
+    final _collectionReferenceMock = CollectionReferenceMock();
+    final _documentReferenceMock = DocumentReferenceMock();
+    final _documentSnapshotMock = DocumentSnapshotMock();
+    final _eventContextMock = EventContextMock();
```

Comment:
Please, don't use the leading underscore for identifiers that are not private. See [Effective Dart](https://dart.dev/guides/language/effective-dart/style#dont-use-a-leading-underscore-for-identifiers-that-arent-private) for more details.